client/src/components/Reviews/components/AddReview.jsx

```diff
@@ -53,33 +58,35 @@ function AddReview({ handleModalToggle, productCharacteristics }) {
   });
 
   function handleReviewSubmit(e) {
-    // TODO add in checks for data
-    // TODO finish data submisson
     e.preventDefault();
+    setSubmissonErr(false);
+    const characteristicSubmit = {};
+    const keysArray = Object.keys(allCharacteristics);
+    keysArray.forEach((key) => {
+      characteristicSubmit[allCharacteristics[key].id] = characteristics[key];
+    });
     const submitedData = {
       ...reviewText,
-      product_id: 'NEED TO CHANGE', // TODO get product id
+      product_id: Number(productId),
       rating,
-      isRecommended,
-      characteristics: {
-        14: characteristics.Size,
-        15: characteristics.Width,
-        16: characteristics.Comfort,
-        17: characteristics.Quality,
-        18: characteristics.Length,
-        19: characteristics.Fit,
-      },
+      recommend: isRecommended,
+      characteristics: characteristicSubmit,
+      photos: [],
     };
-    axios.post('/TEMP', submitedData)
-      .then(() => handleModalToggle)
-      .catch((err) => console.log(err));
+    if (submitedData.email.indexOf('@') === -1 || submitedData.email.indexOf('.') === -1) {
```

Comment:
A simple regex might be a little better here

```
if (submittedData.email.search(/^\S+@\S+\.\S+$/) === -1)
```
`/.../` is used for a regex the way `'...'` is used for a string. Matches:
- `^`: start of string
- `\S`: anything that's not whitespace
- `+`: one or more of previous token, i.e. `\S+` matches one or more non whitespace characters
- `@`: literal @ sign
- `\.`: escaped . (for the .com part)
- `$`: end of string

Although emails can actually include escaped spaces, so this isn't strictly correct, but it's probably good enough for our situation.

Reply:
ah yeah, i need to get more comfortable using regex ty

Reply:
It takes a bit to get comfortable with it, but once you can start doing smaller things with it, it's very useful. [The actual regex used to verify emails is a beast](http://www.ex-parrot.com/~pdw/Mail-RFC822-Address.html).